### DISCUSSION TITLE
X509_LOOKUP_meth_set_get_by_fingerprint

### DISCUSSION THREAD

**Discussion by nagu1137** [openssl/openssl#23894] [Q&A] [answered]
getting error when using X509_LOOKUP_meth_set_get_by_fingerprint.

int x509_w32_get_by_fingerprint(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type, unsigned char *bytes, int len, X509_OBJECT *ret)
{

return 1;
}

X509_LOOKUP_get_by_fingerprint_fn custom_get_by_fingerprint = static_cast<X509_LOOKUP_get_by_fingerprint_fn>(x509_w32_get_by_fingerprint);
int ret7=X509_LOOKUP_meth_set_get_by_fingerprint(x509_w32_lookup,custom_get_by_fingerprint);

cannot convert from 'int (__cdecl *)(X509_LOOKUP *,X509_LOOKUP_TYPE,unsigned char *,int,X509_OBJECT *)' to 'X509_LOOKUP_get_by_fingerprint_fn'
28> None of the functions with this name in scope match the target type

**Comment by t8m** [comment, marked as answer]
You need to use:
```
int x509_w32_get_by_fingerprint(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type, const unsigned char *bytes, int len, X509_OBJECT *ret)
{

return 1;
}
```
